dawn/src/dawn/CMakeLists.txt

```diff
@@ -24,7 +24,8 @@ install(
 # GCC 8, still a common compiler, requires linking to a separate library for std::filesystem support.
 # Once that is no longer common we can remove this hack
 add_library(cpp_extra_libs INTERFACE)
-if (CMAKE_COMPILER_IS_GNUCC AND CMAKE_CXX_COMPILER_VERSION VERSION_LESS 9.0)
+if ((CMAKE_COMPILER_IS_GNUCC OR CMAKE_COMPILER_IS_GNUCXX) AND CMAKE_CXX_COMPILER_VERSION VERSION_LESS 9.0)
```

Comment:
I like all of these changes except `CMAKE_CXX_COMPILER_VERSION VERSION_LESS 9.0`. Here is why:

`libstdc++fs` is necessary for linking to the `Filesystem Technical Specification` [1] that exists in GCC 7 [2]. However, as of GCC 8 [3] this was removed. The relevant part of the CHANGELOG is:

> Runtime Library (libstdc++)
>   - Improved experimental support for C++17, including the following features:
>     - Deduction guides to support class template argument deduction.
>     - std::filesystem implementation.

If this is not the case on a system then it is the fault of the installation, not the compiler itself, and we should get to the bottom of it instead of linking `libstdc++fs` into libraries linked with GCC-8.

I would like to propose merging all but this change for now and investigating further!

[1] https://gcc.gnu.org/onlinedocs/libstdc++/manual/using.html
[2] https://stackoverflow.com/questions/45867379/why-does-gcc-not-seem-to-have-the-filesystem-standard-library
[3] https://gcc.gnu.org/gcc-8/changes.html

Reply:
This appears to be an Ubuntu-specific bug with default library versions. GCC 8 there links to `libstdc++.so.6.0.26`, which does not have the symbols, but later versions seem to include it. The work-around there is to link to `libstdc++fs`, even though it shouldn't be necessary if you read the GCC docs.

Detailed info here: https://bugs.launchpad.net/ubuntu/+source/gcc-8/+bug/1824721

I've changed my mind on this: since most of our Dawn usage is Ubuntu, I think we should merge this.